### PR TITLE
Include file path only

### PR DESCRIPTION
**Pull request by AndreasSchoenle**

### CODE REVIEW

**Comment by AndreasSchoenle** [comment]
Keep full source file path - this is nice for e.g. Debug output sinks. We can rely on the const char * _file_path  to remain valid (__FILE__ macro). 
Possibly one could even move the splitting of the filename to the sink and save the _file member?

**Comment by KjellKod** [comment]
please pull latest from masters so the travis build works

File: src/g3log/logmessage.hpp [hunk 1]
```diff
@@ -73,7 +76,7 @@ namespace g3 {
       LogMessage& operator=(LogMessage other);
 
 
-      LogMessage(const std::string& file, const int line, const std::string& function, const LEVELS& level);
+      LogMessage(const char * file, const int line, const std::string& function, const LEVELS& level);
```
Comment (KjellKod) [inline src/g3log/logmessage.hpp]: Unfortunately you cannot rely on the `char*` to remain valid across library borders. That's why the data is copied into the class. One runtime library can delete the data in a different way than expected in the LogMessage receiving runtime.  

It is *probably* safe to change this but I know it works when copied so please keep the copy paradigm. i.e. keeping it as std::string.

Comment (AndreasSchoenle) [inline src/g3log/logmessage.hpp]: makes sense. will do.

File: src/g3log/logmessage.hpp [hunk 2]
```diff
@@ -87,6 +90,7 @@ namespace g3 {
       timespec _timestamp;
       std::thread::id _call_thread_id;
       std::string _file;
+	  std::string _file_path;
```
Comment (KjellKod) [inline src/g3log/logmessage.hpp]: please adjust formatting

File: src/logmessage.cpp [hunk 3]
```diff
@@ -123,10 +123,11 @@ namespace g3 {
    }
 
 
-   LogMessage::LogMessage(const std::string& file, const int line,
+   LogMessage::LogMessage(const char * file, const int line,
```
Comment (KjellKod) [inline src/logmessage.cpp]: then I think the std::string parameter can be left too as it was originally, right?

Comment (AndreasSchoenle) [inline src/logmessage.cpp]: yes, my bad.

**Comment by KjellKod** [comment]
:+1:

**Pull request merged**